File: ss-front/src/app/model/GeneralInfo.ts
export interface GeneralInfo {
  id: number;
  firstName: string;
  surName: string;
  sex: Sex;
  phone: string;
  birthDate: string;
}

export enum Sex {
  MAN ='Мужчина',
  WOMAN = 'Женщина'
}

export const SexMapping = {
  [Sex.MAN]: "Мужчина",
  [Sex.WOMAN]: "Женщина"
}

File: ss-back/src/main/java/stomat/ssback/model/GeneralInfo.java
package stomat.ssback.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity(name = "general_infos")
public class GeneralInfo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "sur_name")
    private String surName;

    @Column(name = "phone")
    private String phone;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column(name = "sex")
    @Enumerated(EnumType.STRING)
    private Sex sex;

    @Override
    public String toString() {
        return "GeneralInfo{" +
                "id=" + id +
                ", firstName='" + firstName + '\'' +
                ", surName='" + surName + '\'' +
                ", phone='" + phone + '\'' +
                ", birthDate=" + birthDate +
                ", sex=" + sex +
                '}';
    }
}
